pkg/pillar/cmd/domainmgr/rkt.go

```diff
@@ -165,12 +159,17 @@ func ociToRktImageHash(ociFilename string) (string, error) {
 	}
 
 	// if we made it here, we didn't find it, so convert the file to an aci and load it
-	tmpDir, err := ioutil.TempDir("", "acifile")
+	tmpBase := path.Join(types.PersistDir, "tmp")
+	if err := os.MkdirAll(tmpBase, 0700); err != nil {
+		return "", fmt.Errorf("error creating base temporary directory for aci caching %s: %v", tmpBase, err)
+	}
+	defer os.RemoveAll(tmpBase)
```

Comment:
You can't remove all of /persist/tmp since there could be other instances in the process of using that directory.
If you want to make sure you can always remove your own, then prepend with your uuid in /persist/tmp/
Or do a TempDir with the base being /persist/tmp/

Reply:
Shouldn't it be a subdir specific to this task @nordmark @deitch  ? Like /persist/rkt-conversion or something?

Reply:
Done

Reply:
The RemovAll needs to be for tmpDir - the directory you just created. Not tmpBase.

Reply:
You can do a tmpDir, err := iioutil.TempDir(path.Join(types.PersistDir, "tmp"), "rktconversion")
if that is more convenient, but the RemoveAll still needs to be for the unique tmpDir.

Reply:
Put together, your comments @eriknordmark and @rvs make sense. The way this works now, it will create `/persist/tmp/rktconversion`, then create temp dirs underneath for each of the tasks at hand. That actually works well, except in case @eriknordmark highlighted below [here](https://github.com/lf-edge/eve/pull/570#issuecomment-579137301). If we have two concurrent conversions, whichever finishes first will trounce the other as it does a `RemoveAll("/persist/tmp/rktconversion")`.

This just needs to be a tmpdir for this conversion, under which everything else (more tempdirs) will exist.